packages/flutter/lib/src/widgets/gesture_detector.dart

```diff
@@ -309,3 +343,122 @@ class _GestureDetectorState extends State<GestureDetector> {
     }
   }
 }
+
+class _GestureSemantics extends OneChildRenderObjectWidget {
+  _GestureSemantics({
+    Key key,
+    this.onTapDown,
+    this.onTapUp,
+    this.onTap,
+    this.onLongPress,
+    this.onVerticalDragStart,
+    this.onVerticalDragUpdate,
+    this.onVerticalDragEnd,
+    this.onHorizontalDragStart,
+    this.onHorizontalDragUpdate,
+    this.onHorizontalDragEnd,
+    this.onPanStart,
+    this.onPanUpdate,
+    this.onPanEnd,
+    Widget child
+  }) : super(key: key, child: child);
+
+  final GestureTapDownCallback onTapDown;
+  final GestureTapUpCallback onTapUp;
+  final GestureTapCallback onTap;
+  final GestureLongPressCallback onLongPress;
+  final GestureDragStartCallback onVerticalDragStart;
+  final GestureDragUpdateCallback onVerticalDragUpdate;
+  final GestureDragEndCallback onVerticalDragEnd;
+  final GestureDragStartCallback onHorizontalDragStart;
+  final GestureDragUpdateCallback onHorizontalDragUpdate;
+  final GestureDragEndCallback onHorizontalDragEnd;
+  final GesturePanStartCallback onPanStart;
+  final GesturePanUpdateCallback onPanUpdate;
+  final GesturePanEndCallback onPanEnd;
+
+  bool get _watchingTaps {
+    return onTapDown != null
+        || onTapUp != null
+        || onTap != null;
+  }
+
+  bool get _watchingHorizontalDrags {
+    return onHorizontalDragStart != null
+        || onHorizontalDragUpdate != null
+        || onHorizontalDragEnd != null;
+  }
+
+  bool get _watchingVerticalDrags {
+    return onVerticalDragStart != null
+        || onVerticalDragUpdate != null
+        || onVerticalDragEnd != null;
+  }
+
+  bool get _watchingPans {
+    return onPanStart != null
+        || onPanUpdate != null
+        || onPanEnd != null;
+  }
+
+  void _handleTap() {
+    if (onTapDown != null)
+      onTapDown(Point.origin);
+    if (onTapUp != null)
+      onTapUp(Point.origin);
+    if (onTap != null)
+      onTap();
+  }
+
+  void _handleHorizontalDragUpdate(double delta) {
+    if (_watchingHorizontalDrags) {
+      if (onHorizontalDragStart != null)
+        onHorizontalDragStart(Point.origin);
+      if (onHorizontalDragUpdate != null)
+        onHorizontalDragUpdate(delta);
+      if (onHorizontalDragEnd != null)
+        onHorizontalDragEnd(Offset.zero);
+    } else {
+      assert(_watchingPans);
+      if (onPanStart != null)
+        onPanStart(Point.origin);
+      if (onPanUpdate != null)
+        onPanUpdate(new Offset(delta, 0.0));
+      if (onPanEnd != null)
+        onPanEnd(Offset.zero);
+    }
+  }
+
+  void _handleVerticalDragUpdate(double delta) {
+    if (_watchingVerticalDrags) {
+      if (onVerticalDragStart != null)
+        onVerticalDragStart(Point.origin);
+      if (onVerticalDragUpdate != null)
+        onVerticalDragUpdate(delta);
+      if (onVerticalDragEnd != null)
+        onVerticalDragEnd(Offset.zero);
```

Comment:
Looks like you do actually fire the start/end parts.  That seems easier for consumers.